configure.ac

```diff
@@ -18,39 +20,70 @@ License along with this program.  If not, see
 
 ]])
 
-AC_PREREQ(2.52)
-AC_INIT([fibers], [1.1.1])
+AC_PREREQ([2.71])
+AC_INIT([fibers],[1.1.1])
 AC_CONFIG_SRCDIR([env.in])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIRS([m4])
-AM_INIT_AUTOMAKE([serial-tests])
+AM_INIT_AUTOMAKE([subdir-objects serial-tests -Werror foreign])
 AC_COPYRIGHT(FIBERS_CONFIGURE_COPYRIGHT)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
-AC_DISABLE_STATIC
-
-AC_ISC_POSIX
 AC_PROG_CC
-AC_STDC_HEADERS
-AC_PROG_LIBTOOL
 
-AC_CHECK_HEADERS([sys/epoll.h])
-AC_CHECK_FUNCS([epoll_create epoll_create1])
+LT_INIT([disable-static])
 
 AM_CONDITIONAL([CROSS_COMPILING], [test "x$cross_compiling" = "xyes"])
 
-if test "$cross_compiling" = "no"; then
+if test "x$cross_compiling" = "xno"; then
    GUILE_TARGET=""
 else
    GUILE_TARGET="--target=$host_alias"
 fi
 AC_SUBST([GUILE_TARGET])
 
 WARN_CFLAGS=-Wall
-AC_ARG_ENABLE([Werror], AC_HELP_STRING([--disable-Werror],[Don't stop the build on errors]),
-        [], WARN_CFLAGS="-Wall -Werror")
+AC_ARG_ENABLE([Werror], AS_HELP_STRING([--disable-Werror],[Don't stop the build on errors]),
+    [], WARN_CFLAGS="-Wall -Werror")
 AC_SUBST(WARN_CFLAGS)
 
+AC_CHECK_FUNCS(epoll_wait)
+AM_CONDITIONAL([HAVE_EPOLL_WAIT], [test "x$ac_cv_func_epoll_wait" = "xyes"])
+
+AC_ARG_WITH([libevent],
+    [AS_HELP_STRING([--with-libevent],
+        [use libevent for events backend  @<:@default=no@:>@])],
+        [],
+        [with_libevent=no])
+
+AS_IF([test "x$with_libevent" != "xno"],
+    [PKG_CHECK_MODULES([LIBEVENT], [libevent >= 2.1.0],,
+        [AC_MSG_ERROR([unable to find libevent])])])
+AM_CONDITIONAL([WITH_LIBEVENT], [test "x$with_libevent" = "xyes"])
+
+# Detect the target system
+case "$host_os" in
+    *darwin*)
+        needs_clocks_ext=yes
+        needs_affinity_ext=yes
+        PLATFORM=darwin
+        ;;
+    *linux*)
+        needs_clocks_ext=no
+        needs_affinity_ext=no
+        PLATFORM=linux
```

Comment:
With what we have with the last update I believe this is now true. So, we won't build the `affinity` extension if it's available in Guile or we won't build `clocks` extension (for `clock_nanosleep`).

I'm now thinking that maybe there's a way to unify `posix-clocks` into a single file again since most of it is generic. So, we would do:

- Define platform variables (e.g. `darwin?`) in `config.scm.in`.
- Define `needs-clocks-extension?` and `needs-affinity-extension?` or something like that in `config.scm.in` (based on configure.ac). We could detect both in Guile actually but since we already use those to check whether we need to build the extension or not I believe is better to have just one source of truth.
- In `posix-clocks.scm` load only the clocks/affinity extension if those variables are set to true.
- In `posix-clocks.scm` the only thing that currently differs is `pthread-getcpuclockid` so that would just be platform specific using the boolean variables (`darwin?`...).

WDYT?

Reply:
Oh, if you like this, we can rename `_fibers_clock_nanosleep` to `clock_nanosleep`. But I believe that's fine now, because we have detected that it's not available in the system.

Reply:
I am not quite following, but types like suseconds_t, time_t, etc. can have different underlying C types and different underlying int64_t kind of types on different platforms, and I'm not sure that will matter but is good to think about.

Reply:
That's true, then to start maybe we just leave it like it is now and we can simplify in the future.

Reply:
Marking this resolved. For now we will have the two implementations, we can improve later.